data/templates/libkkc-data/tools/sortlm.py

```diff
@@ -143,13 +144,14 @@ def cmp_header(a, b):
             offset += 1
         unigram_file.close()
 
-        print "writing 2-gram file"
+        print("writing 2-gram file")
         bigram_offsets = {}
         bigram_file = open("%s.2gram" % self.__output_prefix, "wb")
-        keys = self.__ngram_entries[1].keys()
+        keys = list(self.__ngram_entries[1].keys())
         items = [(struct.pack("=LL", ids[1], unigram_offsets[ids[0]]), ids) for ids in keys]
         offset = 0
-        for header, ids in sorted(items, cmp=cmp_header):
+        items.sort(key = functools.cmp_to_key(cmp_header))
```

Comment:
This and the `cmp_header` modification look overly complicated. You could simply write: `sorted(items, key=lambda x: x[0])`.